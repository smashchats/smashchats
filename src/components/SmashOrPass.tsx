import React from "react";

import { Pressable } from "react-native";
import { Image } from "expo-image";

import { Colors } from "@/src/constants/Colors.js";
import { NeonText } from "@/src/components/NeonText.jsx";
import { Badge } from "@/src/components/design-system/Badge.jsx";
import { Box } from "@/src/components/design-system/Box.jsx";

const Lightning = require("@/assets/lightning-bolt.png");

type Props = {
    onSmash: () => void;
    onPass: () => void;
};

export function SmashOrPass({ onSmash, onPass }: Readonly<Props>): JSX.Element {
    return (
        <Badge
            borderRadius={16}
            bgColor={Colors.background}
            borderColor={Colors.purple}
            borderWidth={3.5}
        >
            <Pressable testID="SmashOrPass::Smash" onPress={onSmash}>
                <NeonText text="Smash" />
            </Pressable>
            <Box width={25} />
            <Image
                style={{
                    overflow: "visible",
                    position: "absolute",
                    height: 50,
                    width: 25,
                    left: 57,
                    transform: [{ scale: 0.8 }],
                }}
                alt="Lightning bolt"
                source={Lightning}
            />
            <Pressable testID="SmashOrPass::Pass" onPress={onPass}>
                <NeonText text="pass" />
            </Pressable>
        </Badge>
    );
}
