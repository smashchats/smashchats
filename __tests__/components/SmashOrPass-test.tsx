import { fireEvent, render } from "@testing-library/react-native";
import { SmashOrPass } from "@/src/components/SmashOrPass.jsx";

describe("SmashOrPass", () => {
    test("Renders correctly", () => {
        const tree = render(
            <SmashOrPass onSmash={() => {}} onPass={() => {}} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
    test("Can click on pass", () => {
        const fnOnSmash = jest.fn();
        const fnOnPass = jest.fn();
        const { getByTestId } = render(
            <SmashOrPass onSmash={fnOnSmash} onPass={fnOnPass} />
        );
        const pressable = getByTestId("SmashOrPass::Pass");
        fireEvent.press(pressable);

        expect(fnOnSmash).not.toHaveBeenCalled();
        expect(fnOnPass).toHaveBeenCalled();
    });
    test("Can click on smash", () => {
        const fnOnSmash = jest.fn();
        const fnOnPass = jest.fn();
        const { getByTestId } = render(
            <SmashOrPass onSmash={fnOnSmash} onPass={fnOnPass} />
        );
        const pressable = getByTestId("SmashOrPass::Smash");
        fireEvent.press(pressable);

        expect(fnOnSmash).toHaveBeenCalled();
        expect(fnOnPass).not.toHaveBeenCalled();
    });
});
