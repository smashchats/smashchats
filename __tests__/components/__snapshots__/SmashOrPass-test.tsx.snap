// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmashOrPass Renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#222222",
      "borderColor": "#64189b",
      "borderRadius": 16,
      "borderWidth": 3.5,
      "flex": undefined,
      "flexDirection": "row",
      "height": undefined,
      "justifyContent": "center",
      "marginHorizontal": undefined,
      "marginRight": undefined,
      "paddingHorizontal": 8,
      "paddingVertical": 1,
      "width": undefined,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="SmashOrPass::Smash"
  >
    <View>
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "marginHorizontal": 0,
            "marginLeft": undefined,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "textShadowColor": "#64189b",
              "textShadowOffset": {
                "height": 1.1,
                "width": 1.1,
              },
              "textShadowRadius": 1,
              "textTransform": "uppercase",
            }
          }
        >
          Smash
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "marginHorizontal": undefined,
            "marginLeft": 0,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "position": "absolute",
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "textShadowColor": "#64189b",
              "textShadowOffset": {
                "height": -1.1,
                "width": -1.1,
              },
              "textShadowRadius": 1,
              "textTransform": "uppercase",
            }
          }
        >
          Smash
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "marginHorizontal": undefined,
            "marginLeft": 0,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "position": "absolute",
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "textShadowColor": "#64189b",
              "textShadowOffset": {
                "height": 1.1,
                "width": -1.1,
              },
              "textShadowRadius": 1,
              "textTransform": "uppercase",
            }
          }
        >
          Smash
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "marginHorizontal": undefined,
            "marginLeft": 0,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "position": "absolute",
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "textShadowColor": "#64189b",
              "textShadowOffset": {
                "height": -1.1,
                "width": 1.1,
              },
              "textShadowRadius": 1,
              "textTransform": "uppercase",
            }
          }
        >
          Smash
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": undefined,
        "height": undefined,
        "width": 25,
      }
    }
  />
  <ViewManagerAdapter_ExpoImage
    accessibilityLabel="Lightning bolt"
    containerViewRef={"[React.ref]"}
    contentFit="cover"
    contentPosition={
      {
        "left": "50%",
        "top": "50%",
      }
    }
    height={50}
    left={57}
    nativeViewRef={"[React.ref]"}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    overflow="visible"
    placeholder={[]}
    position="absolute"
    source={
      [
        {
          "__packager_asset": true,
          "height": 1,
          "scale": 1,
          "uri": "file://assets/full/path/to/directory/name.type",
          "width": 1,
        },
      ]
    }
    style={
      {
        "height": 50,
        "left": 57,
        "overflow": "visible",
        "position": "absolute",
        "transform": [
          {
            "scale": 0.8,
          },
        ],
        "width": 25,
      }
    }
    transform={
      [
        {
          "scale": 0.8,
        },
      ]
    }
    transition={null}
    width={25}
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="SmashOrPass::Pass"
  >
    <View>
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "marginHorizontal": 0,
            "marginLeft": undefined,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "position": undefined,
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "textShadowColor": "#64189b",
              "textShadowOffset": {
                "height": 1.1,
                "width": 1.1,
              },
              "textShadowRadius": 1,
              "textTransform": "uppercase",
            }
          }
        >
          pass
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "marginHorizontal": undefined,
            "marginLeft": 0,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "position": "absolute",
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "textShadowColor": "#64189b",
              "textShadowOffset": {
                "height": -1.1,
                "width": -1.1,
              },
              "textShadowRadius": 1,
              "textTransform": "uppercase",
            }
          }
        >
          pass
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "marginHorizontal": undefined,
            "marginLeft": 0,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "position": "absolute",
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "textShadowColor": "#64189b",
              "textShadowOffset": {
                "height": 1.1,
                "width": -1.1,
              },
              "textShadowRadius": 1,
              "textTransform": "uppercase",
            }
          }
        >
          pass
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": undefined,
            "flex": undefined,
            "flexDirection": undefined,
            "height": undefined,
            "marginHorizontal": undefined,
            "marginLeft": 0,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "position": "absolute",
            "width": undefined,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
              "fontSize": 14,
              "textShadowColor": "#64189b",
              "textShadowOffset": {
                "height": -1.1,
                "width": 1.1,
              },
              "textShadowRadius": 1,
              "textTransform": "uppercase",
            }
          }
        >
          pass
        </Text>
      </View>
    </View>
  </View>
</View>
`;
